### PR TITLE
OT275-36 : Endpoint para listar Usuarios

### PR DESCRIPTION
**Pull request by LAYALA09**
# [OT275-36#](https://alkemy-labs.atlassian.net/browse/OT275-36)

Endpoint get user listings with sofdelete(false). The body contains firstName, lastName, email, role.

### HOW HAS THIS BEEN TESTED?

Please describe the tests that you ran to verify your changes. Provide instructions so we can
reproduce. Please also list any relevant details for your test configuration.

- [ x] Postman.
- [ ] Unit test.
- [ ] Integration test.
- [ ] No testing required (only applies for tech task).

**Test Configuration**:

* ...
* ...

**Screenshots**:
POSTMAN2

* ....
GET LIST

### CHECKLIST

- [ x] My code follows the style guidelines of this project.
- [ x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have added the new resource in the Postman Collection file.

### CODE REVIEW

File: src/main/java/com/alkemy/ong/infrastructure/config/spring/security/SecurityConfig.java [hunk 1]
```diff
@@ -120,6 +120,10 @@ protected void configure(HttpSecurity http) throws Exception {
         .hasAnyRole(Role.ADMIN.name(), Role.USER.name())
         .antMatchers(HttpMethod.POST, SLIDES_URL)
         .hasRole(Role.ADMIN.name())
+        .antMatchers(HttpMethod.GET, "/users")
+        .hasRole(Role.ADMIN.name())
+        .antMatchers(HttpMethod.PUT, "/activities/{id:[\\d+]}")
+        .hasRole(Role.ADMIN.name())
```
Comment (mkain0) [inline src/main/java/com/alkemy/ong/infrastructure/config/spring/security/SecurityConfig.java:126]: Esto no es parte del scope de este ticket.

Comment (mkain0) [inline src/main/java/com/alkemy/ong/infrastructure/config/spring/security/SecurityConfig.java:126]: Esto no es parte del scope de este ticket.

File: src/main/java/com/alkemy/ong/infrastructure/config/spring/security/SecurityConfig.java [hunk 2]
```diff
@@ -120,6 +120,10 @@ protected void configure(HttpSecurity http) throws Exception {
         .hasAnyRole(Role.ADMIN.name(), Role.USER.name())
         .antMatchers(HttpMethod.POST, SLIDES_URL)
         .hasRole(Role.ADMIN.name())
+        .antMatchers(HttpMethod.GET, "/users")
```
Comment (mkain0) [inline src/main/java/com/alkemy/ong/infrastructure/config/spring/security/SecurityConfig.java:123]: Debería utilizar las constantes.

File: src/main/java/com/alkemy/ong/infrastructure/rest/resource/UserResource.java [hunk 3]
```diff
@@ -24,6 +28,8 @@
 public class UserResource {
 
   private final IDeleteUserUseCase deleteUserUseCase;
+  private final IListUserUseCase listUserUseCase;
+  private final UserMapper userMapper;
```
Comment (mkain0) [inline src/main/java/com/alkemy/ong/infrastructure/rest/resource/UserResource.java:32]: ListUserMapper para ser consistente con el standard definido para este proyecto.

File: src/main/java/com/alkemy/ong/infrastructure/rest/resource/UserResource.java [hunk 4]
```diff
@@ -33,6 +39,13 @@ public ResponseEntity<Void> delete(@PathVariable Long id) {
     return ResponseEntity.status(HttpStatus.NO_CONTENT).build();
   }
 
+  @GetMapping(
+      produces = MediaType.APPLICATION_JSON_VALUE)
+  public ResponseEntity<ListUserResponse> get() {
```
Comment (mkain0) [inline src/main/java/com/alkemy/ong/infrastructure/rest/resource/UserResource.java:44]: get se utiliza para single elements. El nombre correcto sería list()

File: src/main/java/com/alkemy/ong/infrastructure/rest/mapper/UserMapper.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    }
+    UserResponse userResponse = new UserResponse();
+    userResponse.setFirstName(user.getFirstName());
+    userResponse.setLastName(user.getLastName());
+    userResponse.setEmail(user.getEmail());
+    userResponse.setRole(user.getRole().getName());
+    return userResponse;
+  }
+
+  public ListUserResponse toResponse(List<User> userList) {
```
Comment (mkain0) [inline src/main/java/com/alkemy/ong/infrastructure/rest/mapper/UserMapper.java:26]: NAME WITH IMPLEMENTATIONS TYPES: MENCIONADO EN MULTIPLES PR.

File: src/main/java/com/alkemy/ong/infrastructure/rest/mapper/UserMapper.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    if (user == null) {
+      return null;
+    }
+    UserResponse userResponse = new UserResponse();
+    userResponse.setFirstName(user.getFirstName());
+    userResponse.setLastName(user.getLastName());
+    userResponse.setEmail(user.getEmail());
+    userResponse.setRole(user.getRole().getName());
+    return userResponse;
+  }
```
Comment (mkain0) [inline src/main/java/com/alkemy/ong/infrastructure/rest/mapper/UserMapper.java:24]: Esto es responsabilidad de otra clase.

File: src/main/java/com/alkemy/ong/infrastructure/database/repository/abstraction/IUserSpringRepository.java [hunk 7]
```diff
@@ -18,6 +19,9 @@ public interface IUserSpringRepository extends JpaRepository<UserEntity, Long> {
   @Query(value = "SELECT u FROM UserEntity u WHERE u.softDeleted = false AND u.userId = :id")
   Optional<UserEntity> exists(@Param("id") Long id);
 
+  @Query(value = "SELECT u FROM UserEntity u WHERE u.softDeleted = false")
```
Comment (mkain0) [inline src/main/java/com/alkemy/ong/infrastructure/database/repository/abstraction/IUserSpringRepository.java:22]: También se puede resolver con JPA.

File: src/main/java/com/alkemy/ong/infrastructure/database/mapper/UserEntityMapper.java [hunk 8]
```diff
@@ -25,6 +28,17 @@ public User toDomain(UserEntity entity) {
     return user;
   }
 
+  public List<User> toDomain(List<UserEntity> userEntityList) {
```
Comment (mkain0) [inline src/main/java/com/alkemy/ong/infrastructure/database/mapper/UserEntityMapper.java:31]: MISMO COMENTARIO SOBRE LOS NOMBRES.

File: src/main/java/com/alkemy/ong/infrastructure/database/mapper/UserEntityMapper.java [hunk 9]
```diff
@@ -25,6 +28,17 @@ public User toDomain(UserEntity entity) {
...
+    if (userEntityList == null || userEntityList.isEmpty()) {
+      return Collections.emptyList();
+    }
+    List<User> userList = new ArrayList<>(userEntityList.size());
```
Comment (mkain0) [inline src/main/java/com/alkemy/ong/infrastructure/database/mapper/UserEntityMapper.java:35]: MISMO COMENTARIO SOBRE LOS NOMBRES.

**Review by mkain0 — CHANGES_REQUESTED** [review]

**Comment by ezevillafae** [comment]
Hablé con @LAYALA09 y agregué estos cambios, pude encontrar un detalle en el mapper.

**Review by mkain0 — APPROVED** [review]

**Pull request merged**